### DISCUSSION TITLE
How to set default formatter to never use scientific notation using pretty formatter?

### DISCUSSION THREAD

**Discussion by MattTheCuber** [hgrecco/pint#1937] [Q&A]
I am using my formatter with `ureg.default_format = "~P"` at the formatter. This works well, but very small numbers are formatted using scientific notation which is not my preference (e.g. `3.410605131648481×10⁻¹³ ft/s` should be `"0.00000000000034106 ft/s"`).

**Comment by hgrecco** [comment]
With the new formatter delegate, it is much easier to write your own and swap it. The docs are not ready yet, but you can check the https://github.com/hgrecco/pint/blob/master/pint/delegates/formatter/plain.py and https://pint.readthedocs.io/en/develop/user/formatting.html#custom-formats

**Reply by MattTheCuber** [reply]
Hmm, I might take the pretty formatter to customize to get this behavior once its released...

**Comment by keewis** [comment]
for something much simpler than a custom formatter (which do give you a lot of power, but might be overkill here – depends on what you want), you could just use the `f` formatter of standard python floats with a fixed width. For example: `.10f~P`

**Reply by MattTheCuber** [reply]
That could work, but for our case I don't want is print many decimals when not necessary (which is the default behavior).

P.S. `print(ureg.Quantity("0.00000000000034106 ft/s"))` with 10 decimals prints `0.0000000000 ft/s`

~~P.S. 2, this might be a bug currently. I tried setting the `default_format` to `.10f~P`, and then `.10fP`, `.5f`, and many other combinations. But none of them seem to remove the scientific notation. In fact, I tried is with much simpler numbers and nothing seems to be working to use anything with `.Xf`.
cc @hgrecco~~
Nevermind, I was using `__repr__` instead of `__str__` or something because it was in a jupyter notebook.
